### PR TITLE
Delta lake base

### PR DESCRIPTION
**Pull request by MugdhaHardikar-GSLab**
This PR enables user to ingest delta table version history in the form of operation aspect.
`version_history_limit`: this parameter can be set to limit the number of versions pulled. if value is 'n' then 'n' latest version's history will be ingested.
I tried to incorporate few comments from below: https://linear.app/acryl-data/issue/ACR-3094/[delta-lake]-config-validation-has-bugs

## Checklist
- [ ] The PR conforms to DataHub's [Contributing Guideline](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md) (particularly [Commit Message Format](https://github.com/datahub-project/datahub/blob/master/docs/CONTRIBUTING.md#commit-message-format))
- [ ] Links to related issues (if applicable)
- [ ] Tests for the changes have been added/updated (if applicable)
- [ ] Docs related to the changes have been added/updated (if applicable). If a new feature has been added a Usage Guide has been added for the same.
- [ ] For any breaking change/potential downtime/deprecation/big changes an entry has been made in [Updating DataHub](https://github.com/datahub-project/datahub/blob/master/docs/how/updating-datahub.md)

### CODE REVIEW

File: metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py [hunk 1]
```diff
@@ -1,104 +1,12 @@
...
-from datahub.ingestion.source.aws.aws_common import AwsSourceConfig
-from datahub.metadata.schema_classes import GlobalTagsClass, TagAssociationClass
 
 S3_PREFIXES = ["s3://", "s3n://", "s3a://"]
 
 logging.getLogger("py4j").setLevel(logging.ERROR)
 logger: logging.Logger = logging.getLogger(__name__)
 
 
-def get_s3_tags(
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py:16]: Why is this removed ?

File: metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py [hunk 2]
```diff
@@ -122,6 +141,99 @@ def get_fields(self, delta_table: DeltaTable) -> List[SchemaField]:
...
+
+            # History schema picked up from https://docs.delta.io/latest/delta-utility.html#retrieve-delta-table-history
+            reported_time: int = int(time.time() * 1000)
+            last_updated_timestamp: int = hist["timestamp"]
+            statement_type = OPERATION_STATEMENT_TYPES.get(
+                hist.get("operation"), OperationTypeClass.CUSTOM
+            )
+            custom_type = hist.get("operation")
+            operation_custom_properties = dict()
+            operation_custom_properties.update(
```
Comment (mayurinehate) [inline metadata-ingestion/src/datahub/ingestion/source/delta_lake/source.py]: Can we rewrite this update section to improve readability. Issue update only if value is not None.

Something like
```
if hist.get(key1) is not None:
    properties[key1] = hist[key1]
```

OR
```
keys_interested_in = [key1, key2, ...]
for key in hist.keys():
    if key in keys_interested_in and hist[key] is not None:
        properties[key] = hist[key]
```

File: metadata-ingestion/tests/integration/delta_lake/golden_files/local/golden_mces_allow_table.json [hunk 3]
```diff
@@ -514,6 +506,101 @@
...
 },
+{
+    "auditHeader": null,
+    "entityType": "dataset",
+    "entityUrn": "urn:li:dataset:(urn:li:dataPlatform:delta-lake,my-platform.tests/integration/delta_lake/test_data/delta_tables/my_table_no_name,UAT)",
+    "entityKeyAspect": null,
+    "changeType": "UPSERT",
+    "aspectName": "operation",
+    "aspect": {
+        "value": "{\"timestampMillis\": 1615443388097, \"partitionSpec\": {\"type\": \"FULL_TABLE\", \"partition\": \"FULL_TABLE_SNAPSHOT\"}, \"operationType\": \"CUSTOM\", \"customOperationType\": \"WRITE\", \"customProperties\": {\"isBlindAppend\": \"True\", \"isolationLevel\": \"Serializable\"}, \"lastUpdatedTimestamp\": 1655831649166}",
```
Comment (mayurinehate) [inline metadata-ingestion/tests/integration/delta_lake/golden_files/local/golden_mces_allow_table.json]: Doesn't WRITE operation on delta table map cleanly with existing datahub operation types ? Why is it a custom operation ?

File: metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py (hunk 1 above)
Comment (MugdhaHardikar-GSLab) [reply metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py:16]: They are actually moved to the other util class s3_boto_util as a part of another PR. Not sure why these changes are coming here. But i didnt removed the changes as these needs to go away

File: metadata-ingestion/tests/integration/delta_lake/golden_files/local/golden_mces_allow_table.json (hunk 3 above)
Comment (MugdhaHardikar-GSLab) [reply metadata-ingestion/tests/integration/delta_lake/golden_files/local/golden_mces_allow_table.json]: WRITE operation is not existing operation. Standard operations are: 
INSERT = "INSERT"
    
    """Rows were updated"""
    UPDATE = "UPDATE"
    
    """Rows were deleted"""
    DELETE = "DELETE"
    
    """Asset was created"""
    CREATE = "CREATE"
    
    """Asset was altered"""
    ALTER = "ALTER"
    
    """Asset was dropped"""
    DROP = "DROP"
    
    """Custom asset operation"""
    CUSTOM = "CUSTOM"
    
    UNKNOWN = "UNKNOWN"

File: metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py (hunk 1 above)
Comment (mayurinehate) [reply metadata-ingestion/src/datahub/ingestion/source/aws/s3_util.py:16]: Okay, ideally this should go away in final PR.
